File: tests/repo-intel/fixtures/ts-js-app/src/repo/caseRepo.ts
import type { Case } from "../model/Case";
import { CaseStatus } from "../model/Case";

const store = new Map<string, Case>();

export function findById(id: string): Case | undefined {
  return store.get(id);
}

export function save(item: Case): Case {
  store.set(item.id, item);
  return item;
}

export function create(id: string, title: string): Case {
  const item: Case = { id, title, status: CaseStatus.Open };
  return save(item);
}

export async function findByIdAsync(id: string): Promise<Case | undefined> {
  return findById(id);
}

File: tests/repo-intel/fixtures/python-app/acme/repo/case_repo.py
from acme.model.case import Case, CaseStatus

MAX_BATCH = 100


class CaseRepository:
    def find_by_id(self, case_id: int):
        return Case(id=case_id, title="found", status=CaseStatus.OPEN)

    def save(self, case: Case) -> Case:
        return case

    def delete(self, case_id: int) -> None:
        pass
